runtime/caml/mlvalues.h

```diff
@@ -269,6 +273,10 @@ Caml_inline header_t Hd_val(value val)
 /* Fields are numbered from 0. */
 #define Field(x, i) (((volatile value *)(x)) [i]) /* Also an l-value. */
 
+#define Field_atomic(x, i) \
+  atomic_load_explicit(Op_atomic_val((volatile value *)(x) + i), \
+                       memory_order_relaxed)
```

Comment:
This problem for tsan will also affect legacy code written by users. Does this mean that adding `volatile` to `Field` was not enough of a fix in OCaml 5 because this will create false positives for tsan?
The original idea was that a volatile read would behave closely-enough to a relaxed read: is there by chance an option to convince tsan of that?

Reply:
Also, if this is really needed, then some time ago it was discussed that it would be beneficial to split the `Field` macro in two (one l-value and one r-value). This suggest to simply call this macro `Read_field` or `Field_load` or something like that.